backend/globaleaks/handlers/admin/tenant.py

```diff
@@ -193,3 +201,18 @@ def delete(self, tenant_id):
         log.info('Removing tenant with id: %d', tenant_id, tid=self.request.tid)
 
         return delete(tenant_id)
+
+class TenantImportExport(BaseHandler):
+    check_roles = 'admin'
+    invalidate_cache = True
+    root_tenant_only = True
+    invalidate_tenant_states = True
+
+    def get(self, tenant_id):
+        tenant_id = int(tenant_id)
+
+        return export_tenant(tenant_id)
+
+    def post(self):
+        import_tenant(self.request.content.read())
```

Comment:
@NCommander: this need to be handlerd  a file handler where the same way handlers/admin/file.py works.

Here you have a 1GB+ payload hitting you and you are reading it all in RAM.
With a file handler instead the file will arrive in streaming, and here will have just a file descriptor to pass directly to the import_tenant function;
Internally to the import_tenant function we may use the Tar functions to still read it in streaming writing unpacked data on the disk.